packages/samples/headless-react/src/components/sort/sort.class.tsx

```diff
@@ -0,0 +1,80 @@
+import {Component} from 'react';
+import {
+  buildDateSortCriterion,
+  buildFieldSortCriterion,
+  buildNoSortCriterion,
+  buildQueryRankingExpressionSortCriterion,
+  buildRelevanceSortCriterion,
+  buildSort,
+  Sort as HeadlessSort,
+  SortOrder,
+  SortState,
+  Unsubscribe,
+} from '@coveo/headless';
+import {engine} from '../../engine';
+
+const criterions = {
+  Relevance: buildRelevanceSortCriterion(),
+  'Date (Ascending)': buildDateSortCriterion(SortOrder.Ascending),
+  'Date (Descending)': buildDateSortCriterion(SortOrder.Descending),
+  'Size (Ascending)': buildFieldSortCriterion('size', SortOrder.Ascending),
+  'Size (Descending)': buildFieldSortCriterion('size', SortOrder.Descending),
+  Suggested: buildQueryRankingExpressionSortCriterion(),
+  None: buildNoSortCriterion(),
+};
+
+export class Sort extends Component {
+  private controller: HeadlessSort;
+  public state: SortState;
+  private unsubscribe: Unsubscribe = () => {};
+
+  constructor(props: {}) {
+    super(props);
+
+    this.controller = buildSort(engine, {
+      initialState: {criterion: criterions.Suggested},
+    });
+    this.state = this.controller.state;
+  }
+
+  componentDidMount() {
+    this.unsubscribe = this.controller.subscribe(() => this.updateState());
+  }
+
+  componentWillUnmount() {
+    this.unsubscribe();
+  }
+
+  private updateState() {
+    this.setState(this.controller.state);
+  }
+
+  private get criterionNames() {
+    return Object.keys(criterions) as (keyof typeof criterions)[];
+  }
+
+  private sortBy(criterionName: keyof typeof criterions) {
+    this.controller.sortBy(criterions[criterionName]);
+  }
+
+  private isSortedBy(criterionName: keyof typeof criterions) {
+    return this.controller.isSortedBy(criterions[criterionName]);
+  }
+
+  render() {
```

Comment:
I initially tried to build a dropdown, however, React appears to throw errors when using the `selected` attribute of an `option` element. This error specifies that we should be using the `value` or `defaultValue` attribute of `select`, which would be much uglier to implement and wouldn't make use of `isSortedBy`.

Reply:
I personally think a dropdown is a lot more natural of a component, so even if the solution is uglier, we definitely should have an example of it because its more like what a client would want to do. If there's something tricky in getting it to work, I feel like thats even more of reason to have a working example (so clients don't come to us asking how to do this).